libs/portals/core/src/components/PortalProvider.tsx

```diff
@@ -38,18 +48,15 @@ export const PortalProvider = ({
   const activeModule = useMemo(
     () =>
       userInfo
-        ? modules.find((module) =>
-            module
-              // Get all routes for the module
-              .routes({
-                userInfo,
-                client,
-              })
-              // Extract the path from each route
-              .map(({ path }) => path)
-              // Find the route path that matches the current pathname
-              .find((path) => matchPath(path, pathname)),
-          )
+        ? modules.find((module) => {
+            return (
+              spreadRoutsChildren(module.routes({ userInfo, client }))
+                // Extract the path from each route
+                .map(({ path }) => path)
+                // Find the route path that matches the current pathname
+                .find((path) => path && matchPath(path, pathname))
+            )
```

Comment:
Note that this (like other places using `matchPath` does not support relative route paths.

It's tempting to see if we can store extra information on React Router routes (like the portal module) and replace all of our route matching to use `matchRoutes` instead.